src/jit/morph.cpp

```diff
@@ -7671,12 +7671,13 @@ void Compiler::fgMorphTailCall(GenTreeCall* call)
         call->gtFlags |= GTF_EXCEPT;
     }
 
-    // Now inject a placeholder for the real call target that codegen
-    // will generate
-    GenTreePtr arg = new (this, GT_NOP) GenTreeOp(GT_NOP, TYP_I_IMPL);
+// Now inject a placeholder for the real call target that codegen will generate
```

Comment:
Is there a reason that amd64 (and now arm) use a constant zero node rather than a NOP? Or is it just historical? It appears that it is just removed without really looking at it.
Also, you might want to add a comment anchor here, or you could just declare the arg, e.g.:
```
GenTree* arg;
```
and then assign its value in the different arms of the `#ifdef`. IMO that looks a bit better than the ugly anchor

Reply:
> Is there a reason that amd64 (and now arm) use a constant zero node rather than a NOP? Or is it just historical?

Well, the argument is for a call target address, which would be a TYP_I_IMPL, and not TYP_VOID. One assert I was seeing was a type mismatch as a result of the void.

Reply:
Makes sense, thanks.